src/switchroot/ostree-mount-util.h

```diff
@@ -106,6 +120,69 @@ read_proc_cmdline_ostree (void)
   return ret;
 }
 
+static inline char*
+gen_ostree_target (const char* prepend)
+{
+  char dir_str[PATH_MAX];
+  int dir_str_len = snprintf(dir_str, sizeof(dir_str), "%s", prepend);
+  int dir_str_cap = PATH_MAX - dir_str_len;
+  int prepend_len = dir_str_len;
+  dir_str_len += snprintf(dir_str + prepend_len, sizeof(dir_str), "%s", "/ostree");
+  dir_str_cap = PATH_MAX - dir_str_len;
+  char* append_from = dir_str + dir_str_len;
+  int appended_len = 0;
+  DIR __attribute__ ((cleanup(close_dir))) *dir = opendir(dir_str);
+  if (!dir)
+    {
+      fprintf(stderr, "'%s' %d", strerror(errno), __LINE__);
+      return "";
+    }
+
+  struct dirent *ent = 0;
+  for (time_t latest = 0; (ent = readdir(dir)); )
+    {
+      if (ent->d_type == DT_LNK)
+        {
+          struct stat lsb;
+          lsb.st_mtime = 0;
+          lstat(ent->d_name, &lsb);
+          if (latest <= lsb.st_mtime)
+            {
+              appended_len = snprintf(append_from, dir_str_cap, "/%s", ent->d_name);
+              latest = lsb.st_mtime;
+            }
+        }
+    }
+
+  dir_str_len += appended_len;
+  dir_str_cap = PATH_MAX - dir_str_len;
+
+  for (int i = 0; i < 3; ++i)
```

Comment:
What's with the `3` here?  Usually things like this want at least e.g.
```
   const int retries = 3;  // Because...?
```

Reply:
This adds the 3 next dirs to the string, there are 5 directories in the string in total, I can add a comment to make more clear. It's not a retry, it's recursing down the directory.

Reply:
Well it's actually 6 but the first one is "/sysroot" which we need to throw away at the end, we just need it here to open directories because the "/ostree" directory is at "/sysroot/ostree" at this point.

Reply:
This will probably change to something other than, just add 3 next directories onto the end of the string as we figure things out